ruby/red-arrow/lib/arrow/array-builder.rb

```diff
@@ -115,6 +115,17 @@ def detect_builder_info(value, builder_info)
             builder: Date32ArrayBuilder.new,
             detected: true,
           }
+        when BigDecimal
+          if value.precs[0] > Arrow::Decimal128DataType::MAX_PRECISION
```

Comment:
Unfortunately, `value.prec[0]` does not mean the number of digits that `value` has.
It just means the number of digits can be contained in the internal buffer of `value`.

Now we should calculate the number of digits of a BigDecimal from the result of `value.to_s("F")`.

I will fix this issue in bigdecimal.

Reply:
We should add tests of converting from/to `BigDecimal("1e30")` and `BigDecimal("1e70")`. `precs` of them are both `[9, 18]`, but they should be converted to `Decimal128` and `Decimal256`, respectively.

Reply:
Done.